### PR TITLE
added support for the ascii-art library "cowsay"

### PR DESCRIPTION
**Pull request by mojoaxel**
**This is just for fun! 🐇**

1. I updated the dependencies.
2. I added support for the cowsay ASCII-library. 🐄

It is now possible to select ASCII-Images from [cowsay](https://github.com/piuccio/cowsay/tree/master/cows).

Setting `TTC_SAY_BOX` to e.g. `stegosaurus.cow` would result in something like this:
image

All changes are backward compatible - so your current settings of `TTC_SAY_BOX` should continue to work.

### CODE REVIEW

**Comment by mojoaxel** [comment]
Can somebody please review this. Thx!

File: README.md [hunk 1]
```diff
@@ -47,8 +47,9 @@ All the settings the dashboard looks at are in the sample file `sample.env`. Thi
 lists the environment variables that you can copy in your `rc` files:
   - `TTC_BOTS` are the 3 twitter bots to check, comma separated. The first entry
   in this list will be displayed in the big party parrot box.
-  - `TTC_SAY_BOX = parrot | bunny | llama | cat | yeoman | mario | ironman | minions | panda`, to party with a different parrot (or,
-    more specifically: to have a different animal say a message in the big box). You can create your own custom art(.ansi file) [here](https://gauravchl.github.io/ansi-art/webapp/) and download and supply it's absolute path to render it within box. (eg: `TTC_SAY_BOX='/Users/om/desktop/cat.ansi'`)
+  - `TTC_SAY_BOX` defines the ASCII-Art to show.
+    e.g. `parrot` | `bunny` | `llama` | `cat` | `yeoman` | `mario` | `ironman.ansi` | `stegosaurus.cow`
+    See section below for all the crazy options.
```
Comment (notwaldorf) [inline README.md]: Please don't use the word crazy, as it's ableist. How about "for all the different options" :)

File: README.md [hunk 2]
```diff
@@ -65,6 +66,16 @@ lists the environment variables that you can copy in your `rc` files:
...
 
+##### Configure the Say-box
+
+There are almost endless variation of ASCII art images supported.
+You can select an existing image from the [cowsay library](https://github.com/piuccio/cowsay/tree/master/cows) just by defining a filename ending with ".cow". So e.g. `vader.cow` or `bunny.cow` would be two possible options.
+Also the [ansi-art](https://github.com/gauravchl/ansi-art) library is supported.
+In addition to the [existing](https://github.com/gauravchl/ansi-art/tree/master/src/arts) colorful images you can create your own image by using the [webapp](https://gauravchl.github.io/ansi-art/webapp/) and download the result.
+After downloading the ANSI-File you can just supply it's absolute path to render it within box. (eg: `TTC_SAY_BOX='/Users/om/desktop/cat.ansi'`)
+
+In addintion to this libraries the following types are supported: `bunny`, `llama`, `cat`, `yeoman`
```
Comment (notwaldorf) [inline README.md]: typo: addition

File: README.md [hunk 3]
```diff
@@ -65,6 +66,16 @@ lists the environment variables that you can copy in your `rc` files:
   - `TTC_TERMINAL_TITLE` -- set this to false if you don't want the terminal title
   to be changed on startup.
 
+##### Configure the Say-box
+
+There are almost endless variation of ASCII art images supported.
+You can select an existing image from the [cowsay library](https://github.com/piuccio/cowsay/tree/master/cows) just by defining a filename ending with ".cow". So e.g. `vader.cow` or `bunny.cow` would be two possible options.
+Also the [ansi-art](https://github.com/gauravchl/ansi-art) library is supported.
+In addition to the [existing](https://github.com/gauravchl/ansi-art/tree/master/src/arts) colorful images you can create your own image by using the [webapp](https://gauravchl.github.io/ansi-art/webapp/) and download the result.
+After downloading the ANSI-File you can just supply it's absolute path to render it within box. (eg: `TTC_SAY_BOX='/Users/om/desktop/cat.ansi'`)
```
Comment (notwaldorf) [inline README.md]: typo: its absolute path

Comment (notwaldorf) [reply README.md]: typo: within the box

File: care.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ var twitterbot = require(__dirname + '/twitterbot.js');
 var gitbot = require(__dirname + '/gitbot.js');
 var pomodoro = require(__dirname + '/pomodoro.js');
 var ansiArt = require('ansi-art').default;
+var cowsay = require("cowsay");;
```
Comment (notwaldorf) [inline care.js]: nit: single quotes, not double, please. also, there's two `;`

File: care.js [hunk 5]
```diff
@@ -310,19 +312,33 @@ function catSay(text) {
...
-  }
-
-  switch (config.say) {
-    case 'bunny' : return bunnySay(textToSay);
-    case 'llama' : return llamaSay(textToSay);
-    case 'cat'   : return catSay(textToSay);
-    case 'yeoman': return yosay(textToSay);
-    default : return ansiArt.get({ artName: config.say, speechText: textToSay });
+  if (/.ansi$/.test(config.say)) {
+    // if SAY ends with ".ansi" use the 'ansi-art' ligrary
```
Comment (notwaldorf) [inline care.js]: typo: library

File: care.js [hunk 6]
```diff
@@ -310,19 +312,33 @@ function catSay(text) {
...
+        speechText: textToSay
+      });
+    } else {
+      return ansiArt.get({
+        filePath: config.say,
+        speechText: textToSay
+      });
+    }
+  } else if (/.cow$/.test(config.say)) {
+    // if SAY ends with ".cow" use the 'cowsay' ligrary
```
Comment (notwaldorf) [inline care.js]: typo: library

File: care.js [hunk 7]
```diff
@@ -310,19 +312,33 @@ function catSay(text) {
...
-
-  switch (config.say) {
-    case 'bunny' : return bunnySay(textToSay);
-    case 'llama' : return llamaSay(textToSay);
-    case 'cat'   : return catSay(textToSay);
-    case 'yeoman': return yosay(textToSay);
-    default : return ansiArt.get({ artName: config.say, speechText: textToSay });
+  if (/.ansi$/.test(config.say)) {
+    // if SAY ends with ".ansi" use the 'ansi-art' ligrary
+    if (config.say === path.basename(config.say)) {
```
Comment (notwaldorf) [inline care.js]: can we move this into a function called `ansiSay`, to clean it up and make this entire function more readable/

File: README.md (hunk 1 above)
Comment (mojoaxel) [reply README.md]: :+1:

File: care.js (hunk 7 above)
Comment (mojoaxel) [reply care.js]: I'll see what I can do...

Comment (mojoaxel) [reply care.js]: I moved the function `getAnsiArt` and all its dependencies to its own file.
This simplifies `care.js` and hopefully makes it easier for newbies to add new artworks.

@notwaldorf I hope you like it.

**Comment by mojoaxel** [comment]
@notwaldorf I know it looks like a big change but it is not really...I hope you are ok with it.

**Comment by anna-wro** [comment]
@notwaldorf @mojoaxel I'd love to see it in TCC!

File: README.md (hunk 2 above)
Comment (mojoaxel) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (mojoaxel) [reply README.md]: fixed

File: care.js (hunk 4 above)
Comment (mojoaxel) [reply care.js]: fixed

File: care.js (hunk 6 above)
Comment (mojoaxel) [reply care.js]: fixed

File: care.js (hunk 5 above)
Comment (mojoaxel) [reply care.js]: fixed

**Pull request merged**